File: java/hamming/src/main/java/Hamming.java
public class Hamming {

    private String leftStrand;
    private String rightStrand;

    public Hamming(String leftStrand, String rightStrand) {

        int rightSize = rightStrand.length();
        int leftSize = leftStrand.length();

        if (leftSize == 0 && rightSize != 0) {
            throw new IllegalArgumentException("left strand must not be empty.");
        }

        if (leftSize != 0 && rightSize == 0) {
            throw new IllegalArgumentException("right strand must not be empty.");
        }

        if (leftSize != rightSize) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }

        this.leftStrand = leftStrand;
        this.rightStrand = rightStrand;
    }

    public int getHammingDistance() {
        int distance = 0;
        int size = leftStrand.length();

        for (int i = 0; i < size; i++) {
            if (leftStrand.charAt(i) != rightStrand.charAt(i)) {
                distance++;
            }
        }

        return distance;
    }
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        require (leftStrand.length == rightStrand.length){
            "left and right strands must be of equal length"
        }

        return leftStrand.zip(rightStrand).count { it.first != it.second }
    }

    fun compute2(leftStrand: String, rightStrand: String): Int {
        require (leftStrand.length == rightStrand.length){
            "left and right strands must be of equal length"
        }

        val size = leftStrand.length
        var dist = 0

        for (i in 0 until size) {
            if (leftStrand[i] != rightStrand[i]) dist++
        }

        return dist
    }
}
